core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -1365,19 +1369,27 @@ class DAGScheduler(
    */
   private[scheduler] def handleExecutorLost(
       execId: String,
-      filesLost: Boolean,
+      fileLost: Boolean,
+      hostLost: Boolean = false,
+      maybeHost: Option[String] = None,
```

Comment:
I agree the method is very hard to understand. I did some refactoring to make it clearer, let me know what you think.